pandas/tseries/index.py

```diff
@@ -77,16 +77,19 @@ def f(self):
 
             result = tslib.get_start_end_field(values, field, self.freqstr,
                                                month_kw)
+            result = self._maybe_mask_results(result, convert='float64')
+
```

Comment:
And I am also not sure why the `is_leap_year` is treated differently, but converting missing values back to NaN would be an API change, as for some reason that attribute currently keeps it missing values as False:

```
In [14]: idx = pd.DatetimeIndex(['2012-01-01', pd.NaT, '2013-01-01'])

In [15]: idx.is_leap_year
Out[15]: Index([True, False, False], dtype='object')
```